src/account-db.js

```diff
@@ -40,50 +41,214 @@ function hashPassword(password) {
 
 export function needsBootstrap() {
   let accountDb = getAccountDb();
-  let rows = accountDb.all('SELECT * FROM auth');
-  return rows.length === 0;
+  let row = accountDb.first('SELECT count(*) FROM auth');
+  return row['count(*)'] === 0;
 }
 
-export function bootstrap(password) {
+// Supported login settings:
+// "password": "secret_password",
+// "openid": {
+//   "issuer": "https://example.org",
+//   "client_id": "your_client_id",
+//   "client_secret": "your_client_secret",
+//   "server_hostname": "https://actual.your_website.com"
+// }
+export function bootstrap(loginSettings) {
   let accountDb = getAccountDb();
 
-  let rows = accountDb.all('SELECT * FROM auth');
-  if (rows.length !== 0) {
+  if (!needsBootstrap()) {
     return { error: 'already-bootstrapped' };
   }
 
-  if (password == null || password === '') {
-    return { error: 'invalid-password' };
+  if (
+    !loginSettings.hasOwnProperty('password') &&
+    !loginSettings.hasOwnProperty('openid')
+  ) {
+    return { error: 'no-auth-method-selected' };
   }
 
-  // Hash the password. There's really not a strong need for this
-  // since this is a self-hosted instance owned by the user.
-  // However, just in case we do it.
-  let hashed = hashPassword(password);
-  accountDb.mutate('INSERT INTO auth (password) VALUES (?)', [hashed]);
+  if (loginSettings.hasOwnProperty('password')) {
+    const password = loginSettings.password;
+    if (password === null || password === '') {
+      return { error: 'invalid-password' };
+    }
+  }
 
-  let token = uuid.v4();
+  if (loginSettings.hasOwnProperty('openid')) {
```

Comment:
Then the server supports authentication via either method. This is well supported from the server side (and somewhat on client /login although the UI isn't great) but I though it's an uncommon enough configuration not to include it in /bootstrap.

Reply:
I think for now it would be best to only allow one authentication method at once, and then loop back later to add full support for multiple methods if desired.